### PR TITLE
Update kedro catalog create to use new /conf structure

### PR DESCRIPTION
**Pull request by DimedS**
## Description
This Pull Request (PR) refines the Kedro catalog create command. Instead of nesting a new "{pipeline-name}.yml" file in a "/catalog" folder, we now directly create a "catalog_{pipeline-name}.yml" file in the conf folder. This change simplifies the directory structure and improves user accessibility.

## Development notes
I've updated the catalog_path. Now, instead of having the YAML files in a 'catalog' folder, I'm just adding 'catalog_' at the beginning of each filename.

Because of this, I need to tweak some tests the same way. Tests that used to check if the 'catalog' folder exists aren't needed anymore. Also, when I clean up after running tests, I'm no longer deleting the whole 'catalog' folder, just deleting files that start with 'catalog_'. 

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [x] Added tests to cover my changes

### CODE REVIEW

**Review by ankatiyar — APPROVED** [review]
I tested it manually too and it works perfectly! 💯

**Comment by ankatiyar** [comment]
I was wondering what happens when `catalog_{pipeline_name}.yml` already exists? Seems like it'll be overwritten right now.

**Comment by DimedS** [comment]
> I was wondering what happens when `catalog_{pipeline_name}.yml` already exists? Seems like it'll be overwritten right now.

Hi Ankita, thank you for your review. From my perspective, if the file already exists, the function _add_missing_datasets_to_catalog() manages this by first loading the file content, then adding the new missing datasets, and finally saving all of them back. Do you think this approach is acceptable?

File: docs/source/configuration/configuration_basics.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ Kedro merges configuration information and returns a configuration dictionary ac
...
 * If two configuration files contain the same top-level key but are in **different** environment paths (for example, one in `conf/base/`, another in `conf/local/`) then the last loaded path (`conf/local/`) takes precedence as the key value. `ConfigLoader.get` does not raise any errors but a `DEBUG` level log message is emitted with information on the overridden keys.
 
-When using any of the configuration loaders, any top-level keys that start with `_` are considered hidden (or reserved) and are ignored. Those keys will neither trigger a key duplication error nor appear in the resulting configuration dictionary. However, you can still use such keys, for example, as [YAML anchors and aliases](https://www.educative.io/blog/advanced-yaml-syntax-cheatsheet#anchors)
+When using any of the configuration loaders, any top-level keys that start with `_` are considered hidden (or reserved) and are ignored. Those keys will neither trigger a key duplication error nor appear in the resulting configuration dictionary. However, you can still use such keys, for example, as [YAML anchors and aliases](https://www.educative.io/blog/advanced-yaml-syntax-cheatsheet)
```
Comment (stichbury) [inline docs/source/configuration/configuration_basics.md:48]: I fixed this link too in @lrcouto's PR and I fixed it the same way, so this is absolutely fine. Approved!

**Review by stichbury — APPROVED** [review]

**Pull request merged**